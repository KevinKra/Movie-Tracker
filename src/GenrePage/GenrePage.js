import React from "react";

class GenrePage extends React.Component {
  render() {
    return <div>GenrePage component</div>;
  }
}

export default GenrePage;
