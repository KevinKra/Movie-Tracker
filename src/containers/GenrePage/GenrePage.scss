.genre-page{
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-gap: 80px;
  margin: 0 auto;
  justify-items: center;
}