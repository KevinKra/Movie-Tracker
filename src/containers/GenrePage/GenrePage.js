import React from "react";
import "./GenrePage.scss";
import { connect } from "react-redux";
import MovieCard from "../MovieCard/MovieCard";
import PropTypes from "prop-types";

export class GenrePage extends React.Component {
  renderMovieCards = type => {
    return type.map(movie => (
      <MovieCard
        key={movie.id}
        wholeObj={movie}
        name={movie.name}
        title={movie.title}
        img={movie.backdrop_path}
      />
    ));
  };

  render() {
    const { movies, tv, favoriteList, searchResults, genre, isLoggedIn } = this.props;
    let whatToRender;
    const currentPath = this.props.location.pathname;

    if (currentPath === "/Movies") {
      whatToRender = this.renderMovieCards(movies);
    } else if (currentPath === "/TV_Shows") {
      whatToRender = this.renderMovieCards(tv);
    } else if (currentPath === "/Favorites") {
      whatToRender = this.renderMovieCards(favoriteList);
    } else if (currentPath === "/SearchResults") {
      whatToRender = this.renderMovieCards(searchResults);
    } else {
      whatToRender = this.renderMovieCards(genre);
    }

    return (
      <React.Fragment>
        <section className="backgroundImage" />
        {currentPath === "/Favorites" && isLoggedIn === false ? <h1>Please Log in to veiw your favorites</h1> : null}
        {currentPath === "/Favorites" && isLoggedIn && favoriteList.length === 0 ? <h1>Looks like you havent saved any favorites yet!</h1> : null}
        <div className="genre-page">{whatToRender}</div>
      </React.Fragment>
    );
  }
}

GenrePage.propTypes = {
  genre: PropTypes.array.isRequired,
  movies: PropTypes.array.isRequired,
  tv: PropTypes.array.isRequired,
  favoriteList: PropTypes.array.isRequired,
  isLoggedIn: PropTypes.array.isRequired,
  searchResults: PropTypes.array.isRequired
};

const mapStateToProps = state => ({
  genre: state.genre,
  movies: state.movies,
  tv: state.tv,
  favoriteList: state.favoriteList,
  searchResults: state.search,
  isLoggedIn: state.isLoggedIn
});

export default connect(mapStateToProps)(GenrePage);
