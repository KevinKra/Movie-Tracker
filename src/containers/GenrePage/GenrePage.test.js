import React from 'react';
import {GenrePage, mapStateToProps} from './GenrePage';
import { shallow } from 'enzyme';
import MockData from "../../assets/mockData";

let mockMovies = MockData.popularMovies;
let mockShows = MockData.popularShows;
let mockGenre = MockData.upcomingMovies;
let mockFavoriteList = MockData.upcomingMovies;

describe('GenrePage', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<GenrePage genre={mockGenre}
                                       movies={mockMovies}
                                       tv={mockShows}
                                       favoriteList={mockFavoriteList}
                                />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})