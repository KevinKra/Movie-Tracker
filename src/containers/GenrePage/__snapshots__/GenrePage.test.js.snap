// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenrePage should match the snapshot 1`] = `undefined`;
